File: 프로그래머스/lv1/이전/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> map = new HashMap<>();
        for(String s : participant){
            if(map.get(s)!= null){
                map.put(s, map.get(s)+1);
            }else{
                 map.put(s, 1);
            }           
        }
        for(String s : completion){
            int i = map.get(s);
            if(i == 1){
                map.remove(s);
            }else{
                map.put(s, i-1);
            }
        }
        
        for(String key : map.keySet()){
            answer = key;
            break;
        }
            
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    
    dict = {}
    
    for n in participant:
        if dict.get(n) == None:
            dict[n] = 1
        else:
            dict[n] = dict[n] + 1
    
    for comp in completion:
        dict[comp] = dict[comp] - 1

    for name in dict:
        if dict[name] > 0:
            answer = name
    
    return answer